core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -706,10 +709,27 @@ protected File getRootDirFor(String name) {
      * {@link #getActiveInstance()} provides the checked versions of the method.
      * @return The instance. Null if the {@link Jenkins} instance has not been started,
      * or was already shut down
+     * @since 1.653
      */
-    @CLIResolver
     @CheckForNull
+    public static Jenkins getInstanceOrNull() {
+        return HOLDER.getInstance();
+    }
+
+    /**
+     * Gets the {@link Jenkins} singleton. In certain rare cases you may have code that is intended to run before
+     * Jenkins starts or while Jenkins is being shut-down. For those rare cases use {@link #getInstanceOrNull()}.
```

Comment:
Well we should just fix any cases where `cleanUp()` is not exhaustive. 

If you are calling e.g. `Jenkins.getInstance()` or `ExtensionList.lookup` after `cleanUp` has cleared `theInstance` then essentially you will result in a partial re-bootstrap of Jenkins itself and a failure to cleanUp correctly